subsys/bluetooth/host/att.c

```diff
@@ -42,6 +42,10 @@
 #define ATT_CHAN_MAX				1
 #endif /* CONFIG_BT_EATT */
 
+/* Pool for ATT responses */
+NET_BUF_POOL_FIXED_DEFINE(att_rsp_pool, CONFIG_BT_MAX_CONN,
+			  BT_L2CAP_BUF_SIZE(CONFIG_BT_L2CAP_TX_MTU), NULL);
```

Comment:
Do response and confirmation operations need to be `CONFIG_BT_L2CAP_TX_MTU` bytes?
From a position of ignorance I would expect them to only be a few bytes.
I use `CONFIG_BT_L2CAP_TX_MTU=252`, so the 'minimal overhead' starts looking large.

Reply:
In case of confirmation they only need a few bytes, but in case of read response it can get as large as the negotiated MTU so Im afraid we can't really limited that.

You can always reduce the TX buffer size or count if you want to reduce the footprint but at leas with this we garantee that it will never timeout since the application will not be able to allocate from this pool as it will only be used for responses.

Reply:
Alright, I can easily reduce the number of TX buffers given that this issue is fixed

Reply:
The new enhanced ATT protocol allows concurrent transactions to be handled by the stack.

so, here should change to CONFIG_BT_MAX_CONN * ATT_CHAN_MAX ?

Reply:
I wasn't planning to increase the number this many times, so except if you have a strong argument against just waiting the buffers to be released, as it have been using so far, Id keep this to just CONFIG_BT_MAX_CONN. Like I said the application will not be able to hold into this pool so this should be fine as long as the controller dispatches the buffers it should resume at bt_l2cap_create_pdu.

Reply:
> I wasn't planning to increase the number of buffers this many times, so except if you have a strong argument against just waiting the buffers to be released, as it have been using so far, Id keep this to just CONFIG_BT_MAX_CONN. Like I said the application will not be able to hold into this pool so this should be fine as long as the controller dispatches the buffers it should resume at bt_l2cap_create_pdu.